File: test_case_api.go
package main

import (
	"fmt"
	"time"

	"github.com/golang-jwt/jwt/v5"
)

// JWT密钥 - 从配置中获取
const jwtSecret = "your-very-secure-jwt-secret-key-that-is-at-least-32-characters-long"

func generateToken(userID uint, email, role string) (string, error) {
	claims := jwt.MapClaims{
		"user_id": userID,
		"email":   email,
		"role":    role,
		"exp":     time.Now().Add(time.Hour * 1).Unix(),
		"iat":     time.Now().Unix(),
	}

	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
	return token.SignedString([]byte(jwtSecret))
}

func main() {
	fmt.Println("=== 生成测试JWT Token ===")

	// 使用admin用户信息生成token
	token, err := generateToken(1, "admin@law-oa.com", "admin")
	if err != nil {
		fmt.Printf("生成token失败: %v\n", err)
		return
	}

	fmt.Printf("Generated Token: %s\n", token)

	// 保存token到文件
	fmt.Printf("\n测试命令:\n")
	fmt.Printf("curl -X GET \"http://localhost:8080/api/cases\" -H \"Authorization: Bearer %s\" -H \"Content-Type: application/json\" -s | jq '.'\n", token)
	fmt.Printf("\n新建案件测试:\n")
	fmt.Printf(`curl -X POST "http://localhost:8080/api/cases" -H "Authorization: Bearer %s" -H "Content-Type: application/json" -d '{"title":"测试案件","description":"API测试创建的案件","client_id":1,"lawyer_id":2,"case_type":"civil","priority":"medium"}' -s | jq '.'`+"\n", token)
}

File: test_case_api.js
const jwt = require('jsonwebtoken');

// JWT配置（需要与后端一致）
const JWT_SECRET = 'your-very-secure-jwt-secret-key-that-is-at-least-32-characters-long'; // 需要从后端配置获取
const TEST_USER_ID = 4; // admin用户ID
const TEST_USERNAME = 'admin@law-oa.com';
const TEST_ROLE = 'admin';

// 生成测试token
function generateTestToken() {
  const payload = {
    user_id: TEST_USER_ID,
    username: TEST_USERNAME,
    role: TEST_ROLE,
    iat: Math.floor(Date.now() / 1000),
    exp: Math.floor(Date.now() / 1000) + (24 * 60 * 60) // 24小时
  };

  return jwt.sign(payload, JWT_SECRET);
}

// 测试案件API
async function testCaseAPI() {
  const token = generateTestToken();
  console.log('Generated token:', token);

  // 测试案件列表API
  const caseListResponse = await fetch('http://localhost:8080/api/cases', {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json'
    }
  });

  console.log('Case list response status:', caseListResponse.status);
  if (caseListResponse.ok) {
    const caseData = await caseListResponse.json();
    console.log('Case list response:', JSON.stringify(caseData, null, 2));
  } else {
    const errorText = await caseListResponse.text();
    console.log('Case list error:', errorText);
  }

  // 测试案件统计API
  const statsResponse = await fetch('http://localhost:8080/api/cases/stats', {
    method: 'GET',
    headers: {
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json'
    }
  });

  console.log('Stats response status:', statsResponse.status);
  if (statsResponse.ok) {
    const statsData = await statsResponse.json();
    console.log('Stats response:', JSON.stringify(statsData, null, 2));
  } else {
    const errorText = await statsResponse.text();
    console.log('Stats error:', errorText);
  }
}

// 执行测试
testCaseAPI().catch(console.error);